pkgs/by-name/zl/zluta/package.nix

```diff
@@ -0,0 +1,58 @@
+{ stdenv, lib, fetchFromGitHub, rocmPackages, python3, cargo, rustc, cmake, clang, zlib, libxml2, libedit, pkgs }:
+
+pkgs.rustPlatform.buildRustPackage rec {
+  pname = "zluda";
+  version = "3";
+
+  src = fetchFromGitHub {
+    owner = "vosen";
+    repo = "ZLUDA";
+    rev = "v${version}";
+    hash = "sha256-lykM18Ml1eeLMj/y6uPk34QOeh7Y59i1Y0Nr118Manw=";
+    fetchSubmodules = true;
+  };
+
+  cargoPatches = [ ./ZLUDA-cargo.lock.patch ];
+
+  buildInputs = [
+    rocmPackages.clr
+    rocmPackages.miopen
+    rocmPackages.rocm-smi
+    rocmPackages.rocsparse
+    rocmPackages.rocsolver
+    rocmPackages.rocblas
+    rocmPackages.hipblas
+    rocmPackages.rocm-cmake
+    rocmPackages.hipfft
+    python3
+    cargo
+    rustc
+    cmake
```

Comment:
native build input?

Reply:
True. Apart from most of the rocmPackages, there are things that are only needed at build time. Though even for the rocm ones it might make sense to only have them as nativeBuildInputs and let the user decide which ones they want to have, as it depends on what they use zluda for afaik.

Reply:
cmake/rustc/cargo are "native" because they are to be executed on the build platform; rocm libraries are "buildInputs" because they are to be executed on the host platform (potentially different from the build platform)